src/client/datascience/notebookStorage/baseModel.ts

```diff
@@ -210,7 +203,7 @@ export abstract class BaseNotebookModel implements INotebookModel {
     public get onDidEdit(): Event<NotebookModelChange> {
         return this._editEventEmitter.event;
     }
-    public get metadata(): INotebookMetadataLive | undefined {
+    public get metadata(): Readonly<nbformat.INotebookMetadata> | undefined {
         return this.kernelId && this.notebookJson.metadata
             ? {
                   ...this.notebookJson.metadata,
```

Comment:
Wouldn't the id not be necessary here anymore?

Reply:
Nope, its not. We don't need the id at all.
When connecting to remote, all we need is the Uri of the notebook and then the class will get the id based on that Uri. This way we don't need to pass this id around.

Reply:
So shouldn't you be able to remove the 'this.kernelId' from the class?

Reply:
Yup removed, i thought i removed it, removed for certain now.
Also our existing functional tests will ensure this is tested correctly.